client/src/client.rs

```diff
@@ -351,6 +351,22 @@ pub trait RpcApi: Sized {
         self.call("getmininginfo", &[])
     }
 
+    fn get_block_template(
+        &self,
+        mode: json::GetBlockTemplateOptionsModes,
+        rules: Vec<json::GetBlockTemplateOptionsRules>,
+        capabilities: Vec<json::GetBlockTemplateOptionsCapabilities>,
+    ) -> Result<json::GetBlockTemplateResult> {
+        self.call(
+            "getblocktemplate",
+            &[into_json(json::GetBlockTemplateOptions {
```

Comment:
This intermediary type can be defined in this method. There's no need for that type to be exported.

You can do just a simple

```rust
#[derive(Serialize)]
struct Argument {
    mode: json::GetBlockTemplateMode,
    rules: Vec<json::GetBlockTemplateRule,
    ...
}
```